brains/andalite.py

```diff
@@ -16,34 +17,34 @@ class Andalite(Sapiens):
     def __init__(self, game, args=None, server='ws://localhost:8765'):
         Sapiens.__init__(self, game, args)
         self.telepathy = Telepathy(server)
+        # get sync class for the currently loaded game.
+        game_sync_class = game_sync_classes.get_game_sync_class(game.emu.gameinfo['name'])
+        if game_sync_class != None:
+            self.game_sync = game_sync_class(self.readMemory, self.writeMemory)
 
     def Step(self):
-        self.sendMemory()
-        self.receiveBytes()
+        if self.game_sync != None:
+            to_send = self.game_sync.on_emulator_step(self.getReceivedData())
+            if len(to_send) > 0:
```

Comment:
this if check could probably just be spelled "if to_send" and it might also be better placed inside of sendData rather than around its invocation?  (unless it makes sense to call sendData with an empty to_send elsewhere in the codebase)

Reply:
k